src/convertToObject.js

```diff
@@ -3,16 +3,39 @@
 /**
  * Implement convertToObject function:
  *
- * Function takes a string with styles (see an example in [stylesString.js](./stylesString.js))
+ * Function takes a string with styles
+ * (see an example in [stylesString.js](./stylesString.js))
  * and returns an object where CSS properties are keys
- * and values are the values of related CSS properties (see an exampl in [test file](./convertToObject.test.js))
+ * and values are the values of related CSS properties
+ * (see an exampl in [test file](./convertToObject.test.js))
  *
  * @param {string} sourceString
  *
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const partsString = sourceString.split(';');
+  const newPartsArr = partsString.map(changesParts);
+  const result = {};
+
+  for (const item of newPartsArr) {
+    Object.assign(result, item);
+  }
+
+  return result;
+}
+
+function changesParts(part) {
```

Comment:
you can use it with forEach instead of a map and add new values straight to result object. You don't need 2 cycles